tests/ctest/test_cmerge.c

```diff
@@ -189,6 +200,77 @@ static void test_order (char * our_order, char * their_order, char * base_order,
 	keyDel (informationKey);
 }
 
+/**
+ * Checks if adding a single line to an array gives one error as happens in regular diff algorithms
+ * or as many errors as there are number of lines, which a trivial implementation does.
+ *
+ *       ours their base result
+ *  /#0  0    0     a
+ *  /#1  1    1     0
+ *  /#2  2    2     1
+ *  /#3  3    3     2
+ *  /#4             3
+ */
+static void array_conflict_number_test (void)
+{
+	printf ("Executing %s\n", __func__);
+	Key * our_root = keyNew ("user/our", KEY_END);
+	Key * their_root = keyNew ("user/their", KEY_END);
+	Key * base_root = keyNew ("user/base", KEY_END);
+	Key * result_root = keyNew ("user/result", KEY_END);
+	Key * informationKey = keyNew (0, KEY_END);
+	KeySet * our = ksNew (5, keyNew ("user/our/#0", KEY_VALUE, "a", KEY_END), keyNew ("user/our/#1", KEY_VALUE, "0", KEY_END),
+			      keyNew ("user/our/#2", KEY_VALUE, "1", KEY_END), keyNew ("user/our/#3", KEY_VALUE, "2", KEY_END),
+			      keyNew ("user/our/#4", KEY_VALUE, "3", KEY_END), KS_END);
+	KeySet * their =
+		ksNew (4, keyNew ("user/their/#0", KEY_VALUE, "0", KEY_END), keyNew ("user/their/#1", KEY_VALUE, "1", KEY_END),
+		       keyNew ("user/their/#2", KEY_VALUE, "2", KEY_END), keyNew ("user/their/#3", KEY_VALUE, "3", KEY_END), KS_END);
+	KeySet * base = ksNew (4, keyNew ("user/base/#0", KEY_VALUE, "0", KEY_END), keyNew ("user/base/#1", KEY_VALUE, "1", KEY_END),
+			       keyNew ("user/base/#2", KEY_VALUE, "2", KEY_END), keyNew ("user/base/#3", KEY_VALUE, "3", KEY_END), KS_END);
+	KeySet * result =
+		elektraMerge (our, our_root, their, their_root, base, base_root, result_root, MERGE_STRATEGY_ABORT, informationKey);
+	// Key * resultKey = ksLookupByName (result, "user/result/#0", 0);
```

Comment:
commented out?

Reply:
I continued this PR in #3128. There I have had those comments already removed. I just cherry-picked the relevant commit into this PR.